doc/md/knowledge-center/mysql/generated_columns.md

```diff
@@ -0,0 +1,218 @@
+
+Virtual Generated Columns are not stored, and only evaluated when a row is read _(after BEFORE [triggers](https://dev.mysql.com/doc/refman/5.7/en/trigger-syntax.html))_. As a result, virtual generated columns take no storage at the cost of CPU cycles for `SELECT` statements.
+
+### Limitations of Generated Columns
+
+Generated Column expressions must be deterministic which means that — given the same input — an expression must always produce the same output. As a result, Generated Columns can not be used with stored variables, functions, procedures, and subqueries; which could cause the output to be non-deterministic. Following this constraint, Generated Columns can not be used to generate random values. On the other hand, a Generated Column may reference other generated columns within the same table, as long as those columns are declared before the Generated Column. In contrast, a Generated Column may reference any non-generated column _regardless_ of its position within the table.
+
+## When to use Generated Columns?
+
+Generated columns should be used whenever you want to create a column with a value that can be directly determined from the values of other columns in the same row. In simpler words, for data that is dependent on other data. This saves the developer from complex application code that is prone to errors on `SELECT`, `INSERT` and `UPDATE` statements. It also ensures that data which must be consistent, stays consistent.
+
+**MySQL Syntax for a Generated Column**
+
+```sql
+column_name data_type [GENERATED ALWAYS] AS (expr) [VIRTUAL | STORED] 
+    [NOT NULL | NULL] [UNIQUE [KEY]] [[PRIMARY] KEY] [COMMENT 'string']
+```
+
+### Using Stored Generated Columns
+
+Stored Generated Columns should be used for data _(in a table)_ that is read more frequently than it is updated. This saves CPU cycles while reading rows _(via `SELECT`)_. Stored Generated Columns should also be used when you want to index the column or use it as a foreign key constraint. Alternatively, use stored generated columns as a cache for complex conditions that are costly to calculate.
+
+#### Example
+
+The following example declares a Stored Generated Column in a table that stores the base and height of a triangle in the `base` and `height` column, then computes its area in `area` _(when triangles are inserted or updated)_.
+
+```sql
+CREATE TABLE triangles (
+    base   DOUBLE,
+    height DOUBLE,
+    area   DOUBLE AS (base * height * 1/2) STORED
+);
+```
+
+### Using Virtual Generated Columns
+
+Virtual Generated Columns should be used for computed data _(in a table)_ that is updated more frequently than it is read or computed data that is expensive to store on disk _(via `INSERT` or `UPDATE`)_. Since values are calculated on the fly, virtual generated columns are perfect for table columns that will have a new value for every `SELECT` statement. If you use the _InnoDB Storage Engine_, secondary indexes can be defined on virtual columns.
```

Comment:
```suggestion
Virtual generated columns should be used for computed data _(in a table)_ that is updated more frequently than it is read or computed data that is expensive to store on disk _(via `INSERT` or `UPDATE`)_. Since values are calculated on the fly, virtual generated columns are perfect for table columns that will have a new value for every `SELECT` statement. If you use the _InnoDB Storage Engine_, secondary indexes can be defined on virtual columns.
```